tests/src/Functional/DeveloperDoesNotExistTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\Tests\apigee_edge\Functional;
+
+use Symfony\Component\HttpFoundation\Response;
+
+class DeveloperDoesNotExistTest extends ApigeeEdgeFunctionalTestBase {
+
+  /**
+   * User prefix.
+   *
+   * @var string
+   */
+  protected $prefix;
+
+  /**
+   * Drupal user.
+   *
+   * @var \Drupal\user\Entity\User
+   */
+  protected $drupalUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->prefix = $this->randomMachineName();
+    _apigee_edge_set_sync_in_progress(TRUE);
+    $this->drupalUser = $this->createAccount([], TRUE, $this->prefix);
+    $this->drupalUser->save();
+    _apigee_edge_set_sync_in_progress(FALSE);
+  }
+
+  /**
+   * Tests an unavailable developer.
+   */
+  public function testFail() {
+    $this->drupalLogin($this->drupalUser);
+    $paths = [
```

Comment:
`$routes = [`
`  'entity.developer_app.canonical_by_developer',`
`  'entity.developer_app.add_form_for_developer',`
`  ...`
`]`
`$route_parameters = [`
` 'user' => $this->drupalUser->id(),`
`]`
`...`
`$this->drupalGet(Url::fromRoute($route, $route_parameters));`